plugins/repository-s3/src/main/java/org/elasticsearch/repositories/s3/S3BlobStore.java

```diff
@@ -105,8 +109,10 @@ public void collectMetrics(Request<?> request, Response<?> response) {
     private long getRequestCount(Request<?> request) {
         Number requestCount = request.getAWSRequestMetrics().getTimingInfo()
             .getCounter(AWSRequestMetrics.Field.RequestCount.name());
-        assert requestCount != null;
-
+        if (requestCount == null) {
+            logger.warn("Expected request count to be tracked for request [{}] but found not count.", request);
```

Comment:
This has never happened on Cloud ever and I believe the original issue reporting this is likely some error with the credentials or an S3 SDK bug (can't reproduce it) but I still think we should log it as we really only get here if there's a problem.

Reply:
Yes, that seems strange. I took a quick glance to the SDK code and it seems like `AWSRequestMetrics` is created before the request is executed 🤔 as long as the collector is enabled for that request.

Reply:
> seems like AWSRequestMetrics is created before the request is executed thinking 

The problem is that the counter is only there if an actual request was executed. Now if there is some bug (like we had before with an NPE in some URL utility method in the SDK) or legitmate exception (maybe from auth/signing) before that we never get the expected counter itself.